habitat/core/dataset.py

```diff
@@ -355,9 +392,57 @@ def _shuffle_iterator(self) -> None:
                 list(g)
                 for k, g in groupby(self._iterator, key=lambda x: x.scene_id)
             ]
+
             random.shuffle(grouped_episodes)
+            for i in range(len(grouped_episodes)):
```

Comment:
Maybe use same approach to shuffling as in init method:
```
        random.shuffle(self.episodes)
        if group_by_scene:
            self.episodes = sorted(self.episodes, key=lambda x: x.scene_id)
```

Reply:
Yeah, makes sense to do that here instead.  The more complicated shuffle logic to be `O(n)` instead of `O(n log(n))` was worth it when it was being called on every swap, but this now gets called very rarely.